exercises/hamming/hamming_test.go

```diff
@@ -2,8 +2,16 @@ package hamming
 
 import "testing"
 
+// targetTestVersion is used to ensure that a solution is only evaluated
+// against the correct version of the test suite.
```

Comment:
I moved this explanation of the test versioning from the `leap` exercise, so that it is explained in a core exercise rather than a bonus one. See https://github.com/exercism/go/pull/808#discussion_r134322951 for my reasoning.